recipes/libraw/all/conanfile.py

```diff
@@ -25,10 +37,21 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
     def requirements(self):
-        self.requires("libjpeg/9d")
-        self.requires("lcms/2.11")
-        self.requires("jasper/2.0.16")
+        # TODO: RawSpeed dependency (-DUSE_RAWSPEED)
+        # TODO: DNG SDK dependency (-DUSE_DNGSDK)
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9d")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.1")
```

Comment:
Can it use libjpeg-turbo out of the box?
You haven't changed any arguments to the cmake script.
Shouldn't the cmake script receive a hint what library to select? To ensure reproducibility.

Reply:
libjpeg-turbo is generally libjpeg compatible. From their hompage: "libjpeg-turbo implements both the traditional libjpeg API as well as the less powerful but more straightforward TurboJPEG API". libjpeg has previously made some bad ABI decisions described [here](https://libjpeg-turbo.org/About/Jpeg-9), but they do not affect libraw jpeg usage.

Generally libjpeg-turbo should always be preferred as it is faster in most cases. libraw is one of the few packages left in Conan that does not have an option to choose which implementation to use.

Reply:
I'm asking because the libjpeg and libjpeg-turbo recipes create different cmake modules.

Reply:
Ah, I see. We link against `${CONAN_LIBS}` which in theory should contain all deps. Since libjpeg-turbo has `provides = "libjpeg"` there should be only one library active at a time as far as I understand.

For `add_definitions(-DUSE_JPEG -DUSE_JPEG8)` I do check for both libraries with `if (TARGET CONAN_PKG::libjpeg OR TARGET CONAN_PKG::libjpeg-turbo)`.

Having said all that there are some CI build failures due to linking. In some cases all three linked libraries cannot be found. In another case lcms2 library cannot be found on Windows. Not really sure what is going on.